tl-services/src/main/java/org/egov/tl/service/PaymentUpdateService.java

```diff
@@ -122,7 +138,13 @@ public void process(HashMap<String, Object> record) {
 					updateRequest.getLicenses()
 							.forEach(obj -> log.info(" the status of the application is : " + obj.getStatus()));
 
-					enrichmentService.postStatusEnrichment(updateRequest);
+					List<String> endStates = Collections.nCopies(updateRequest.getLicenses().size(), STATUS_APPROVED);
```

Comment:
To make a list of size equal to license array with 'STATUS_APPROVED' as default value. This is to identify when the license number should be generated. In TL this state is "APPROVED" so we are creating list using nCopies. In case of BPA  each value will come from MDMS.